analysis_options.yaml

```diff
@@ -0,0 +1,21 @@
+# This is the default analysis_options.yaml for all packages and examples
+# If a package needs specific rules, create the analysis_options.yaml on it
+# include this file, like:
+#
+# include: ../analysis_options.yaml
+#
+# TODO (albertus-stripe): Change this to all_lint_rules.yaml for stricter lint and better codebase going forward after all documentation on public members are done.
+include: package:pedantic/analysis_options.1.11.0.yaml
```

Comment:
I think pedantic options is quite a good one here, ensuring the code quality of our codebases remains high without being too strict and slowing us down in the long term. But happy to discuss about this changes if any of you have different opinion

Reply:
I would personally go for the new https://github.com/dart-lang/lints package.

Reply:
@jamesblasco Yes sure! I think pedantic would be a subset of this package. Would you be ok with us temporarily using `pedantic` for now and re-evaluating this once we have agreement and decision on which lint?
Changing this on this PR would make the PR even bigger than it is right now.